src/App.js

```diff
@@ -1,17 +1,201 @@
+      playerDiceRolls: [0, 0],
+      currentPlayerScore: 0,
+      diceOrder: 1,
+      audio: new Audio(audioClips[2].sound), // set background music
+      isMusicOn: false, // background music is initially OFF
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleContinue = this.handleContinue.bind(this);
+  }
+
+  handleChange(event) {
+    const { name, value } = event.target;
+    this.setState({
+      [name]: parseInt(value),
+    });
+  }
+
+  handleSubmit(event) {
+    const { name, value } = event.target;
+    event.preventDefault();
+    this.state.audio.play(); // start playing background music
+    this.state.audio.loop = true;
+
+    this.setState((previousState) => ({
+      [name]: value,
+      userScores: Array(this.state.numberOfPlayers).fill(0),
+      gameStarted: true,
+      isMusicOn: true,
+    }));
+  }
+
+  handleContinue(event) {
+    event.preventDefault();
+    if (!this.state.userRollDice) {
+      soundPlay(audioClips[0].sound, false, 1); // play Dice shake - louder
+      this.setState((previousState) => ({
+        currentPlayer: nextPlayer(
+          previousState.currentPlayer,
+          this.state.numberOfPlayers
+        ),
+        playerDiceRolls: [rollDice(), rollDice()],
+        userRollDice: !previousState.userRollDice,
+        userOrderDice: false,
+      }));
+    } else {
+      let userScore =
+        this.state.diceOrder === 1
+          ? this.state.playerDiceRolls[0] * 10 + this.state.playerDiceRolls[1]
+          : this.state.playerDiceRolls[1] * 10 + this.state.playerDiceRolls[0];
+      const newScores = this.state.userScores.slice(); //copy the array
+      newScores[this.state.currentPlayer - 1] += userScore; //add new score to copied array location
+      this.setState((previousState) => ({
+        userScore: userScore, // current player score
+        userScores: newScores, // store player score in array
+        currentRound: nextRound(
+          previousState.currentRound,
+          previousState.currentPlayer
+        ),
+        userRollDice: !previousState.userRollDice,
+        userOrderDice: !previousState.userOrderDice,
+      }));
+    }
+  }
+
+  resetGame = () => {
+    this.setState({
+      // Reset the whole game to let user play again
+    });
+  };
+
+  toggleMusic = (event) => {
+    event.preventDefault();
+    let isMusicOn = this.state.isMusicOn;
+    if (isMusicOn) {
+      this.state.audio.pause(); // Pause music if it is playing
+    } else {
+      this.state.audio.play(); // Play music if it is paused
+    }
+    this.setState((previousState) => ({
+      isMusicOn: !previousState.isMusicOn,
+    }));
+  };
+
   render() {
+    const gameStarted = this.state.gameStarted;
+    const userRollDice = this.state.userRollDice;
+    const userOrderDice = this.state.userOrderDice;
+    const playerDiceRolls = this.state.playerDiceRolls;
+    const userScores = this.state.userScores;
```

Comment:
Same here, no need to create new variables. If you absolutely want to, use destructuring for nicer syntax!
```suggestion
const { gameStatrted, userRollDice, userOrderDice, playerDiceRolls, userScores } = this.state;
```